src/catalog/catalog.cpp

```diff
@@ -325,6 +328,31 @@ void CatalogContent::readFromFile(const std::string& directory, DBFileType dbFil
     SerDeser::deserializeValue<table_id_t>(nextTableID, fileInfo.get(), offset);
 }
 
+ExpressionType CatalogContent::getFunctionType(const std::string& name) const {
+    auto upperCaseName = StringUtils::getUpper(name);
+    if (builtInVectorFunctions->containsFunction(upperCaseName)) {
+        return FUNCTION;
+    } else if (builtInAggregateFunctions->containsFunction(upperCaseName)) {
+        return AGGREGATE_FUNCTION;
+    } else if (macros.contains(upperCaseName)) {
+        return MACRO;
```

Comment:
We should probably do a refactor of ExpressionType at some point. All these 3 things seem to be of `FUNCTION` `EXPRESSION TYPE` but 3 different `FUNCTION TYPE`. Open an issue for me maybe

Reply:
open issue #1836